R/tmb-methods.R

```diff
@@ -610,15 +615,14 @@ simulate.mmrm_tmb <- function(object,
   method <- match.arg(method)
 
   # Ensure new data has the same levels as original data.
-  full_frame <- model.frame(
-    object,
-    data = newdata,
-    include = c("subject_var", "visit_var", "group_var", "response_var"),
-    na.action = "na.pass"
+  newdata <- h_factor_ref_data(
+    newdata,
+    object$tmb_data$full_frame,
+    setdiff(all.vars(object$formula_parts$formula[[3]]), object$formula_parts$subject_var)
```

Comment:
this line also seems duplicate, would it maybe be better to also handle this in the helper function? or another small one?